### ISSUE TITLE
building on osx

### ISSUE THREAD

**Issue by deployable** [pjps/ndjbdns#17]
Hello,

Had a couple of issues building on osx (Darwin Kernel Version 14.5.0) with both xcode and brew gcc

The `#ifndef __USE_GNU` section is used when building. 

`getline` is already defined on osx and an error is thrown to that effect

```
common.c:72:1: error: conflicting types for 'getline'
 getline (char **lineptr, ssize_t *n, FILE *stream)
 ^
In file included from common.c:28:0:
/usr/include/stdio.h:442:9: note: previous declaration of 'getline' was here
 ssize_t getline(char ** __restrict, size_t * __restrict, FILE * __restrict) __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_4_3);
         ^
common.c: In function 'read_conf':
common.c:208:41: warning: pointer targets in passing argument 2 of 'getline' differ in signedness [-Wpointer-sign]
     while ((signed)(n = getline (&line, &l, fp)) != -1)
                                         ^
common.c:72:1: note: expected 'ssize_t * {aka long int *}' but argument is of type 'size_t * {aka long unsigned int *}'
 getline (char **lineptr, ssize_t *n, FILE *stream)
 ^
common.c: In function 'gettimezone':
common.c:327:20: warning: implicit declaration of function 'memrchr' [-Wimplicit-function-declaration]
         if (!(p2 = memrchr (tzbuf, '\n', st.st_size)))
                    ^
common.c:327:18: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
         if (!(p2 = memrchr (tzbuf, '\n', st.st_size)))
                  ^
common.c:329:18: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
         if (!(p1 = memrchr (tzbuf, '\n', st.st_size - 1)))
                  ^
Makefile:1150: recipe for target 'common.o' failed
gmake[2]: *** [common.o] Error 1
gmake[2]: Leaving directory '/Users/matt/build/ndjbdns-1.06'
Makefile:1252: recipe for target 'all-recursive' failed
gmake[1]: *** [all-recursive] Error 1
gmake[1]: Leaving directory '/Users/matt/build/ndjbdns-1.06'
Makefile:718: recipe for target 'all' failed
gmake: *** [all] Error 2
```

When the getline definition is removed the build continues on until 
`memrchar` as it doesn't exist on bsd. 

```
/usr/local/Cellar/gcc/5.2.0/bin/gcc-5 -DHAVE_CONFIG_H -I.    -g -O2 -Wall -Wextra -fstack-check -fPIC -D PREFIX=\"/usr/local\" -D SYSCONFDIR=\"/usr/local/etc\" -g -O2 -MT common.o -MD -MP -MF .deps/common.Tpo -c -o common.o common.c
common.c: In function 'gettimezone':
common.c:302:20: warning: implicit declaration of function 'memrchr' [-Wimplicit-function-declaration]
         if (!(p2 = memrchr (tzbuf, '\n', st.st_size)))
                    ^
common.c:302:18: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
         if (!(p2 = memrchr (tzbuf, '\n', st.st_size)))
                  ^
common.c:304:18: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
         if (!(p1 = memrchr (tzbuf, '\n', st.st_size - 1)))
                  ^
mv -f .deps/common.Tpo .deps/common.Po
/usr/local/Cellar/gcc/5.2.0/bin/gcc-5 -g -O2 -Wall -Wextra -fstack-check -fPIC -D PREFIX=\"/usr/local\" -D SYSCONFDIR=\"/usr/local/etc\" -g -O2   -o axfrdns axfrdns.o iopause.o droproot.o tdlookup.o response.o log.o prot.o timeoutread.o timeoutwrite.o common.o libdns.a libtai.a liballoc.a libenv.a libcdb.a libbuffer.a libunix.a libbyte.a 
Undefined symbols for architecture x86_64:
  "_memrchr", referenced from:
      _gettimezone in common.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
Makefile:916: recipe for target 'axfrdns' failed
gmake[2]: *** [axfrdns] Error 1
gmake[2]: Leaving directory '/Users/matt/build/ndjbdns-1.06'
Makefile:1261: recipe for target 'all-recursive' failed
gmake[1]: *** [all-recursive] Error 1
gmake[1]: Leaving directory '/Users/matt/build/ndjbdns-1.06'
Makefile:728: recipe for target 'all' failed
gmake: *** [all] Error 2
```

I had to add: https://code.google.com/p/wchar-utf16/source/browse/src/string/memrchr.c?r=75631a8728eefea484c7856c892889781b426e32 to the #ifndef __USE_GNU section and it builds ok.

**Comment by deployable** [comment]
I guess a define for OSX around the getline definition work?

**Comment by deployable** [comment]
https://github.com/pjps/ndjbdns/pull/20

**Comment by pjps** [comment]
Hello Matt,

Thank you so much for reporting this issue and sending in the pull request. I really appreciate it. The fix patch looks okay at glance, I'll build with it soon. Sorry about the delayed response here. I've seen the other issues too, will respond to them at the earliest.

Thank you.

**Comment by msimerson** [comment]
Any reason not to close these issues once you've merged a PR?

Also, GitHub would do have automaticallly done it for you if you had put the magic term, "Fixes #17" on it's own line in the description of #20
